app/javascript/Pages/Home.jsx

```diff
@@ -1,37 +1,208 @@
-import React, { Component } from "react";
+import React from "react";
+import axios from "axios";
+import PropTypes from "prop-types";
+import { useState, useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { Head } from "@inertiajs/inertia-react";
 import Navbar from "./components/Navbar/Navbar";
 import Footer from "./components/Footer/Footer";
-import { Head, Link } from "@inertiajs/inertia-react";
-import "../stylesheets/Home.css";
-
-class Homepage extends Component {
-  render() {
-    return (
-      <main>
-        <Navbar current_user={this.props.current_user} />
-
-        <img
-          src={require("../images/banner.svg")}
-          alt=""
-          role="presentation"
-          className="hero-img"
-        />
-        <Head title="Welcome" />
-        <p>Hello, welcome to your first Inertia app!</p>
-        <p className="paragraph">This is cool!</p>
-
-        <Link
-          href="/logout"
-          method="delete"
-          headers={window.defaultHeaders}
-          as="button"
-        >
-          Logout
-        </Link>
-        <Footer />
-      </main>
-    );
-  }
+import SocialCard from "./components/SocialCard";
+import CardCategory from "./components/CardCategory";
+import CardPopular from "./components/CardPopular";
+import ButtonFilter from "./components/ButtonFilter";
+
+function Homepage(props) {
+  const { t } = useTranslation();
+  const [items, setItems] = useState([]);
+  const [cats, setCat] = useState([]);
+  const [pclasses, setPClasses] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [categoryFilter, setCategoryFilter] = useState("");
+  const [difFilter, setDifFilter] = useState("beginner");
+  const dif = ["beginner", "intermediate", "advanced"];
+
+  const fetchClassesDifficulty = () => {
+    let newClassData = [];
+    axios
+      .get(`/api/classes?difficulty=${difFilter}`)
+      .then((response) => {
+        newClassData = response.data;
+        setItems(newClassData);
+      })
+      .catch((error) => {
+        console.log(error);
+        newClassData = [];
+      });
+  };
+
+  const fetchCategories = async () => {
```

Comment:
**issue (blocking)**: there are still some functions with async that don't need it